django/contrib/admin/sites.py

```diff
@@ -491,6 +493,7 @@ def get_app_list(self, request):
         return app_list
 
     @never_cache
+    @login_not_required
```

Comment:
I don't think this is correct: the admin index page does require login no? Certainly I'm redirected to login on current versions...

Reply:
Yes, that's true. But when admin user try to browse `www.site.com/admin/` , mapped view is `index`. So that this middleware redirects this user back to the user login page `/accounts/login/?next=/admin/` which is different than the admin login page `/admin/login/?next=/admin/`.

I intentionally used `login_not_required` annotation for admin index view so that the middleware pass the request. When the AdminSite class initiated, it redirects user back to admin login url. Otherwise admin user will be redirected to the users login page.

If you have any suggestion to overcome this problem without calling `login_not_required` for admin index view, please share it <3